app/src/main/java/fi/thl/koronahaavi/MainActivity.kt

```diff
@@ -121,6 +141,16 @@ class MainActivity : AppCompatActivity() {
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
         resolutionViewModel.handleActivityResult(requestCode, resultCode)
+
+        resolutionViewModel.whitelistResolvedEvent().observe(this, Observer {
```

Comment:
Hi Denzil, this livedata observation should probably be setup in onCreate.. but we will work on that when integrating these changes, thanks for the work on this!